Feature/ablunit coverage

PeterJudgeZA opened the pull request:
New properties to support code coverage (on/off) and the folder into which to write profiler output.

Test updates for same, in AblUnitTest.groovy. 

Plugin version bump to 0.2.9.

Commit:
Add support for code coverage to ABLUnit

New properties to support code coverage (on/off) and the folder into which to write profiler output.
Test updates for same.

Commit:
Add tests for profiler calls from the task

Add tests for profiler calls from the task

zerovian commented on src/main/groovy/oe/espresso/latte/AblUnit.groovy:
```diff
@@ -390,10 +395,21 @@ class AblUnit extends BaseLatteSourceTask {
+                if (!profilerOutputDir) {
+                    profilerOutputDir = new File(project.buildDir, "profiler")
+                }
+                ant.Profiler([enabled     : true,
```
I think this needs a profilerOutputDir.mkdirs() so the user doesn't have to do it manually.

PeterJudgeZA replied:
It my testing the folder is created, without needing to call `mkdirs()` . I can certainly add it in.

Commit:
Ensure profiler dir exists; tweak desc

Make sure that the output directory exists before the run.
Tweak the profiler description to be useful (only show the name)

zerovian commented on src/main/groovy/oe/espresso/latte/AblUnit.groovy:
```diff
@@ -401,8 +401,11 @@ class AblUnit extends BaseLatteSourceTask {
                     profilerOutputDir = new File(project.buildDir, "profiler")
                 }
+                // make sure the directory exists
+                project.fileTree(profilerOutputDir)
```
this doesn't create physical files on the disk.  this is a "getter" function that builds an in memory representation of the files that may already be on disk

https://docs.gradle.org/current/javadoc/org/gradle/api/file/FileTree.html

zerovian replied:
yes. but "filetree" is "in memory" not "on disk".  this won't modify the file system.  since you need PCT to write a file to disk, the directory needs to exist,  if PCT is already creating the folder then you don't need to do this, but i've seen many cases where PCT doesn't actually create the folder (it is lazy).

zerovian replied:
also filetree() function returns an object...but your code just throws away that object.

zerovian replied:
I believe this is what you want:

https://docs.gradle.org/current/javadoc/org/gradle/api/Project.html#mkdir-java.lang.Object-

PeterJudgeZA replied:
Updated to use `File.mkdirs`

PeterJudgeZA commented:
Per https://docs.gradle.org/current/userguide/working_with_files.html#sec:locating_files , which says
The simplest way to create a file tree is to pass a file or directory path to the Project.fileTree(java.lang.Object)<https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:fileTree(java.lang.Object)> method. This will create a tree of all the files and directories in that base directory (but not the base directory itself).




From: Matt Baker <notifications@github.com>
Sent: Wednesday, 27 May, 2020 13:07
To: progress/latte <latte@noreply.github.com>
Cc: Peter Judge <pjudge@progress.com>; Author <author@noreply.github.com>
Subject: Re: [progress/latte] Feature/ablunit coverage (#30)


@zerovian requested changes on this pull request.

________________________________

In src/main/groovy/oe/espresso/latte/AblUnit.groovy<https://github.com/progress/latte/pull/30#discussion_r431302865>:

> @@ -401,8 +401,11 @@ class AblUnit extends BaseLatteSourceTask {

                 if (!profilerOutputDir) {

                     profilerOutputDir = new File(project.buildDir, "profiler")

                 }

+                // make sure the directory exists

+                project.fileTree(profilerOutputDir)

this doesn't create physical files on the disk. this is a "getter" function that builds an in memory representation of the files that may already be on disk

https://docs.gradle.org/current/javadoc/org/gradle/api/file/FileTree.html<https://docs.gradle.org/current/javadoc/org/gradle/api/file/FileTree.html>

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/progress/latte/pull/30#pullrequestreview-419433880>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AB7IJ2OAR4WCIHWRYGE7FLLRTVCBRANCNFSM4NLEBXBA>.

Commit:
Force-create the folder structure

Force-create the folder structure
